BaikalDataGenerator/Source/main.cpp

```diff
@@ -20,24 +20,30 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 ********************************************************************/
 
-#include "cline_parser.h"
+#include "cmd_line_parser.h"
 #include "render.h"
 
-void Run(DGenConfig config)
+void Run(const DGenConfig& config)
 {
-    auto render = Render::Create(config.scene_file, config.scene_dir);
+    Render render(config.scene_file, config.width, config.height);
 
-    render->LoadMaterialXml(config.material_dir, config.material_file);
-    render->LoadLightXml(config.light_dir, config.light_file);
-    render->LoadCameraXml(config.camera_dir, config.camera_file);
-    render->GenerateDataset(config.output_dir);
+    if (!config.material_file.empty())
```

Comment:
We should check existence of arguments earlier (during the parsing) and raise the corresponding exception

Reply:
removed